module_utils/network/junos/facts/facts.py

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright 2019 Red Hat
+# GNU General Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+"""
+The facts class for junos
+this file validates each subset of facts and selectively
+calls the appropriate facts gathering function
+"""
+
+from ansible.module_utils.six import iteritems
+from ansible.module_utils.network. \
+    junos.argspec.facts.facts import FactsArgs
+from ansible.module_utils.network. \
+    junos.argspec.l3_interfaces.l3_interfaces import L3_interfacesArgs
+from ansible.module_utils.network. \
+    junos.facts.base import FactsBase
+from ansible.module_utils.network. \
+    junos.facts.l3_interfaces.l3_interfaces import L3_interfacesFacts
+
+
+FACT_SUBSETS = {}
+
+
+class Facts(FactsArgs, FactsBase):  # pylint: disable=R0903
+    """ The fact class for junos
+    """
+
+    VALID_GATHER_SUBSETS = frozenset(FACT_SUBSETS.keys())
+
+    @staticmethod
+    def gen_runable(module, subsets, valid_subsets):
```

Comment:
This needs to be refactored as per the updated changes in ansible/ansible and RMB builder code